src/tests/cwrapper/test_cwrapper.c

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+#include "cwrapper.h"
+
+void test_cwrapper() {
+    char* s;
+    basic x = basic_new(), y = basic_new(), z = basic_new();
+    symbol_set(x, "x");
+    symbol_set(y, "y");
+    symbol_set(z, "z");
+    
+    s = basic_str(x);
+    printf("Symbol : %s\n", s);
+    basic e = basic_new();
+
+    integer_set_ui(e, 123);
+    s = basic_str(e);
+    printf("Integer : %s\n", s);
+
+    integer_set_ui(e, 456);
+    basic_add(e, e, x);
+    basic_mul(e, e, y);
+    basic_div(e, e, z);
+    printf("Basic : %s\n", basic_str(e));
+    basic_diff(e, e, z);
+    printf("Basic : %s\n", basic_str(e));
+
+    rational_set_ui(e, 100, 47);
+    s = basic_str(e);
+
+    printf("Rational : %s\n", s);
+    printf("Is_a_Symbol %s: %d\n", s, is_a_Symbol(e));
+    printf("Is_a_Rational %s: %d\n", s, is_a_Rational(e));
+    printf("Is_a_Integer %s: %d\n", s, is_a_Integer(e));
```

Comment:
Yes, `x, y, z` should be deallocated. 
I also checked for `e`, although I only used print statements. `e` should be deallocated once at the end. Thanks for checking it with valgrind